activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/MessageProducerActivitiEventListener.java

```diff
@@ -24,10 +29,34 @@ public MessageProducerActivitiEventListener(ProcessEngineChannels producer,
 
     @Override
     public void onEvent(ActivitiEvent event) {
+        CommandContext currentCommandContext = Context.getCommandContext();
         ProcessEngineEvent newEvent = converterContext.from(event);
-        if (newEvent != null) {
-            producer.auditProducer().send(MessageBuilder.withPayload(newEvent).build());
+        if (newEvent == null) {
+            return;
+        }
+        @SuppressWarnings("unchecked")
+        List<ProcessEngineEvent> events = (List<ProcessEngineEvent>) currentCommandContext
+                                                                                          .getAttribute(MessageProducerCommandContextCloseListener.PROCESS_ENGINE_EVENTS);
+        if (events != null) {
+            events.add(newEvent);
+        } else {
+            events = new ArrayList<ProcessEngineEvent>();
+            events.add(newEvent);
+            currentCommandContext.addAttribute(MessageProducerCommandContextCloseListener.PROCESS_ENGINE_EVENTS,
+                                               events);
+        }
+
+        List<CommandContextCloseListener> listeners = currentCommandContext.getCloseListeners();
```

Comment:
I'd move all this block inside a method `hasCloseListener(listener)` inside the `CommandContext` then here you can simplify to something like:
``` java
if (!currentCommandContext.hasCloseListener(MessageProducerCommandContextCloseListener.class)) {
    MessageProducerCommandContextCloseListener messageListener = new MessageProducerCommandContextCloseListener(
                                                                                                                     producer);
    currentCommandContext.addCloseListener(messageListener);
}
```
By the way, you can convert `MessageProducerCommandContextCloseListener` to a `@Compoment` and inject it in this bean: the `producer` can be removed from here then. The code above becomes:

```
if (!currentCommandContext.hasCloseListener(MessageProducerCommandContextCloseListener.class)) {
    currentCommandContext.addCloseListener(messageListener);
}
```